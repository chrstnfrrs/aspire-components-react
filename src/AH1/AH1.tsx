import * as React from 'react';

import * as TypographySelectors from '../selectors/typography-selectors';

const AH1: React.FC<TypographySelectors.IHeading> = ({
  as,
  'data-testid': testid,
  type,
  ...props
}) => {
  const componentProps = {
    ...props,
    as: as || 'h1',
    'data-testid': testid,
    type: type || 'h1',
  };

  const { Component, className } = TypographySelectors.getElement(
    componentProps,
  );

  return (
    <Component className={className} data-testid={testid}>
      {props.children}
    </Component>
  );
};

export { AH1 };
