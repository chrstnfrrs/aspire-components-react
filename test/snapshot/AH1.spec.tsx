import { AH1 } from '../../src/AH1/AH1';
import { AS_HEADING_ENUM } from '../enums/as-heading-enum';
import { FONTSIZE_ENUM } from '../enums/fontsize-enum';
import { TYPE_HEADING_ENUM } from '../enums/type-heading-enum';

import { testComponent } from './utils/test-component';
import { testTypography } from './utils/test-typography';

describe('Given AH1 component is used', () => {
  const AH1Props = {
    factory: {
      fontSize: FONTSIZE_ENUM,
    },
  };

  testComponent(AH1, 'AH1', AH1Props);
  testTypography(AH1, 'AH1', 'as', AS_HEADING_ENUM);
  testTypography(AH1, 'AH1', 'type', TYPE_HEADING_ENUM);
});
