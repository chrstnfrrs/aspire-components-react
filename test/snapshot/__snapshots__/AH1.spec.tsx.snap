// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AH1 component is used When a developer uses AH1 with a className Then the the class should be present 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When a developer uses AH1 with no props Then the only class should be a-container 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 2xl Then the class fontSize-2xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 3xl Then the class fontSize-3xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 4xl Then the class fontSize-4xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 5xl Then the class fontSize-5xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 6xl Then the class fontSize-6xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 7xl Then the class fontSize-7xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 8xl Then the class fontSize-8xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 9xl Then the class fontSize-9xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in an invalid value Then no fontSize class should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in base Then the class fontSize-base should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in lg Then the class fontSize-lg should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in sm Then the class fontSize-sm should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in undefined Then no fontSize class should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xl Then the class fontSize-xl should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xs Then the class fontSize-xs should be used 1`] = `<h1 />`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h1" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h2" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h3" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h4" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h5" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with as set to "h6" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h1" Then the component should render as expected 1`] = `
<h1>
  AH1
</h1>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h2" Then the component should render as expected 1`] = `
<h2>
  AH1
</h2>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h3" Then the component should render as expected 1`] = `
<h3>
  AH1
</h3>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h4" Then the component should render as expected 1`] = `
<h4>
  AH1
</h4>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h5" Then the component should render as expected 1`] = `
<h5>
  AH1
</h5>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "h6" Then the component should render as expected 1`] = `
<h6>
  AH1
</h6>
`;

exports[`Given AH1 component is used When rendering AH1 with type set to "strong" Then the component should render as expected 1`] = `
<strong>
  AH1
</strong>
`;
